syntax = "proto3";
package health_assessment_rule;




import "resource_monitor_sdk/model/health_assessment/health_assessment_event_score_config_item.proto";
import "resource_monitor_sdk/model/health_assessment/health_assessment_related_resource_score_config_item.proto";

/*
ListHealthAssessmentRule请求
*/
message ListHealthAssessmentRuleRequest {
    /*
    页码
    */
    int32 page = 1 ;
    /*
    页大小
    */
    int32 pageSize = 2 ;
}

/*
ListHealthAssessmentRule返回
*/
message ListHealthAssessmentRuleResponse {
message List {
    /*
    实例ID
    */
    string id = 1 ;
    /*
    所属模型id
    */
    string objectId = 2 ;
    /*
    关联名称
    */
    string name = 3 ;
    /*
    关联模型Id
    */
    string curVersionId = 4 ;
    /*
    事件评分设置
    */
    repeated health_assessment.HealthAssessmentEventScoreConfigItem eventScoreConfig = 5 ;
    /*
    关联资源评分设置
    */
    repeated health_assessment.HealthAssessmentRelatedResourceScoreConfigItem relatedResourceScoreConfig = 6 ;
    /*
    事件评分总权重
    */
    int32 eventScoreWeight = 7 ;
    /*
    关联资源总权重
    */
    int32 relatedResourceWeight = 8 ;
    /*
    原始指标Id
    */
    string originalMetricId = 9 ;
}
        
    
    /*
    页数
    */
    int32 page = 1 ;
    /*
    页大小
    */
    int32 page_size = 2 ;
    /*
    总数
    */
    int32 total = 3 ;
    /*
    数据列表
    */
    repeated List list = 4 ;
}

/*
ListHealthAssessmentRuleApi返回
*/
message ListHealthAssessmentRuleResponseWrapper {
    /*
    返回码
    */
    int32 code = 1 ;
    /*
    返回码解释
    */
    string codeExplain = 2 ;
    /*
    错误详情
    */
    string error = 3 ;
    /*
    返回数据
    */
    health_assessment_rule.ListHealthAssessmentRuleResponse data = 4 ;
}

